libutils/ip_address.c

```diff
@@ -105,6 +105,32 @@ static int IPV4_parser(const char *source, struct IPV4Address *address)
         address->port = 0;
     }
 
+    {
+        // shortcut for the 0a0b0c0d format
+        unsigned int a, b, c, d, pport = 0;
+        if (strlen(source) == 8 &&
+            4 == sscanf(source, "%2x%2x%2x%2x", &a, &b, &c, &d))
+        {
+            address->octets[3] = a;
+            address->octets[2] = b;
+            address->octets[1] = c;
+            address->octets[0] = d;
+            return 0;
+        }
+
+        // shortcut for the 0a0b0c0d:0e0f format
+        if (strlen(source) == 8+1+4 &&
+            5 == sscanf(source, "%2x%2x%2x%2x:%4x", &a, &b, &c, &d, &pport))
+        {
+            address->octets[3] = a;
+            address->octets[2] = b;
+            address->octets[1] = c;
+            address->octets[0] = d;
+            address->port = pport;
+            return 0;
+        }
+    }
+
```

Comment:
Both of these functions (`IPV4_parser` and `IPV6_parser`) state their expected input, which is the text format. I think we should not magically accept hex format behind people's backs. Since you already know that the string is a hex string of a certain length, there is no way it can be an invalid address if it passes the `sscanf`, so I'd just make a new function, `IPAddressNewFromHex()` or similar to perform this parsing step.